lib/_http_client.js

```diff
@@ -789,14 +789,14 @@ function onSocketNT(req, socket, err) {
       if (!err && req.agent && !socket.destroyed) {
         socket.emit('free');
       } else {
-        finished(socket.destroy(err || req[kError]), (er) => {
-          _destroy(req, er || err);
+        finished(socket.destroy(chainErrors(req[kError], err)), (er) => {
+          _destroy(req, chainErrors(err, er));
```

Comment:
Please don't change stream related stuff before changing the core streams.

Reply:
Not sure what you mean by `changing the core streams`, so I've removed the changes from the PR.